src/__experimental__/components/date/date.component.js

```diff
@@ -46,18 +46,50 @@ class BaseDateInput extends React.Component {
       this.input.focus();
       this.openDatePicker(true);
     }
+    this.handleValidationUpdate(this.inputProps());
   }
 
   componentDidUpdate(prevProps) {
     if (this.isControlled && !this.inputHasFocus && this.hasValueChanged(prevProps)) {
       this.updateSelectedDate(this.props.value);
     }
+
+    if (this.hasValidationsChanged()) {
+      this.handleValidationUpdate(this.inputProps());
+    }
+  }
+
+  inputProps = () => {
+    const { minDate, maxDate, ...inputProps } = this.props;
+    return inputProps;
   }
 
   hasValueChanged = (prevProps) => {
     return this.props.value && prevProps.value !== this.props.value;
   };
 
+  hasValidationsChanged = () => {
+    const { validationsArray } = this.state;
+
+    const currentValidations = concatAllValidations(this.inputProps());
+
+    if (validationsArray.length !== currentValidations.length) {
+      return true;
+    }
+
+    for (let i = validationsArray.length; i--;) {
+      if (validationsArray[i] !== currentValidations[i]) {
+        return true;
+      }
+    }
+
+    return false;
+  }
+
+  handleValidationUpdate = (inputProps) => {
+    this.setState({ validationsArray: concatAllValidations(inputProps) });
+  }
```

Comment:
Will there ever be a situation where `handleValidationUpdate()` is passed anything other than `this.inputProps()` ?

If not, then maybe `handleValidationUpdate()` could be rewritten like this?
```js
handleValidationUpdate = () => {
  const inputProps = this.inputProps();
  this.setState({ validationsArray: concatAllValidations(inputProps) });
}
```
and then it can be called like this (with no arguments):
```js
this.handleValidationUpdate();
```

Reply:
Yeah very good point I'll refactor it 👍

Reply:
Thanks for refactoring that in 38d0e2a :white_check_mark: